### PR TITLE
[10.0][MIG][web_shortcut] 

### PR DESCRIPTION
**Pull request by tarteo**

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Looks good to me, expect the bugs that are also in the 8.0 version

File: web_shortcut/models/web_shortcut.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class WebShortcut(models.Model):
+    _name = 'web.shortcut'
+
+    name = fields.Char('Shortcut Name', size=64)
+    menu_id = fields.Many2one('ir.ui.menu', ondelete='cascade')
+    user_id = fields.Many2one('res.users', 'User Ref.', required=True,
+                              ondelete='cascade', index=True,
+                              default=lambda obj, cr, uid, context: uid)
```
Comment (adrienpeiffer) [inline web_shortcut/models/web_shortcut.py]: ```lambda self: self.env.user.id```

**Comment by ThomasBinsfeld** [comment]
@tarteo Hi, thanks for the merges. I'm working on bringing the compatibility with Odoo 10.0 Enterprise. I'll submit a PR soon.

**Comment by astirpe** [comment]
I'm experiencing an incompatibility with module "board".

Steps to reproduce:
- On a new database, install module `board`
- Open any view and add it to My Dashboard (click on Favourites)
- Install module `web_shortcut`
- Open My Dashboard, following error occurs:


```
Odoo Client Error

TypeError: this.action_manager.webclient is undefined

http://127.0.0.1:8069/web/content/271-b248ffe/web.assets_backend.js:2637
Traceback:
.shortcut_check@http://127.0.0.1:8069/web/content/271-b248ffe/web.assets_backend.js:2637:44
.switch_mode/<@http://127.0.0.1:8069/web/content/271-b248ffe/web.assets_backend.js:2635:2608
jQuery.Callbacks/fire@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:541:281
jQuery.Callbacks/self.fireWith@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:546:198
.Deferred/</deferred[tuple[0]]@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:548:31

...

jQuery.Callbacks/fire@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:541:281
jQuery.Callbacks/self.fireWith@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:546:198
done@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:937:86
.send/callback@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:957:15
```

**Comment by pedrobaeza** [comment]
@Yajo, can you help @astirpe?

**Comment by astirpe** [comment]
Yes, the incompatibility with module "board" is fixed to me. :+1:  Thanks @tarteo !

**Review by astirpe — APPROVED** [review]

**Review by yajo — APPROVED** [review]
Code review OK.

I never really understood why would anybody use this addon instead of just adding some favourites directly in the browser, 🤔 but I'm not gonna block it for that.

File: web_shortcut/static/src/js/web_shortcut.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ odoo.define('web.shortcut', function (require) {
         WebClient = require('web.WebClient'),
         ViewManager = require('web.ViewManager'),
         ActionManager = require('web.ActionManager'),
+        Menu = require('web.Menu'),
```
Comment (ThomasBinsfeld) [inline web_shortcut/static/src/js/web_shortcut.js]: @tarteo That dependency makes the module unusable on Enterprise version.

Comment (tarteo) [reply web_shortcut/static/src/js/web_shortcut.js]: @ThomasBinsfeld Sorry, I can't spend anymore time on this.

Comment (JonathanNEMRY) [reply web_shortcut/static/src/js/web_shortcut.js]: hello @tarteo 
we spent a lot of time on this one too and with your last adding feature (https://github.com/OCA/web/pull/565/commits/4faa3bf6e09a1272bba41b54aa3752634612ab78) it's like reverting our contribution as this commit implies an inconsistency with enterprise version.
As you have no more time I suggest you to add this functionality into another module (event a local module to you) that inherit of this one. Many Thanks in advance

**Review by JonathanNEMRY — CHANGES_REQUESTED** [review]
Do not use require('web.Menu') as it is not enterprise-compatible

File: web_shortcut/static/src/js/web_shortcut.js (hunk 2 above)
Comment (tarteo) [reply web_shortcut/static/src/js/web_shortcut.js]: I reverted my changes to keep it compatible with Enterprise. My changes were to fix some bugs.

Comment (JonathanNEMRY) [reply web_shortcut/static/src/js/web_shortcut.js]: Was it required to add web.Menu to fix those bugs? Can you point me the 'reintroduced' bug by this revert?

Comment (tarteo) [reply web_shortcut/static/src/js/web_shortcut.js]: - When the page is refreshed on a favourite view the toggle is not 'checked'. And other cases that the toggle is not in the correct state.
- The toggle blinks every time the view changes

**Review by adrienpeiffer — CHANGES_REQUESTED** [review]

**Review by JonathanNEMRY — APPROVED** [review]

File: web_shortcut/models/web_shortcut.py (hunk 1 above)
Comment (JonathanNEMRY) [reply web_shortcut/models/web_shortcut.py]: @tarteo Ok now, just the last tips of @adrienpeiffer  and GO for merge :smile:

**Review by ThomasBinsfeld — APPROVED** [review]

File: web_shortcut/models/web_shortcut.py (hunk 1 above)
Comment (tarteo) [reply web_shortcut/models/web_shortcut.py]: Thanks @adrienpeiffer, I changed it :)

**Comment by tarteo** [comment]
Can we merge?

**Comment by pedrobaeza** [comment]
Can you please squash a bit your commits?

**Comment by tarteo** [comment]
@pedrobaeza done :)

**Comment by pedrobaeza** [comment]
I still see 35 commits. Isn't this a lot for this module?

**Comment by tarteo** [comment]
@pedrobaeza I wasn't able to rebase, so i've created a new branch and pr. https://github.com/OCA/web/pull/626

**Pull request closed without merging**